### DISCUSSION TITLE
How to geocode the velocity.h5 and timeseries.h5 obtained from hyp3+mintpy.

### DISCUSSION THREAD

**Discussion by DD-ZHAO** [insarlab/MintPy#1297] [Q&A]
Hello everyone. The interferograms generated by hyp3 using ASF Vertex and using mintpy for time-series analysis resulted in velocity.h5 and timeseries.h5 that cannot be geocoded using geocode.py due to the lack of a lookup table. Using Python scripts to extract key value pairs from .h5 can obtain numerical values for velocity and deformation, but lacks geographic coordinates, so it is not possible to map the velocity externally (such as using GMT or GIS software). How can I geocode the velocity.h5 and timeseries.h5 obtained from hyp3+mintpy. Thank you for your reply.

**Comment by david-ncu2019** [comment]
dear @DD-ZHAO,

the reason you cannot apply `geocode.py` on the HYP3 product because it's already been geocoded in advance.

**Reply by DD-ZHAO** [reply]
Dear @david-ncu2019 
Thank you very much for your reply! I already know that hyp3 has encoded the interferograms geographically, but there is no coordinate system information in the velocity.h5 file after completing the time series analysis. How can I obtain a velocity map with geographic coordinates after hyp3+mintpy processing flow.

**Reply by david-ncu2019** [reply]
Dear @DD-ZHAO,

Because we have both `geometry.h5` file and `velocity.h5` file, both have the same size, it is really possible to extract point-by-point measurement along with coordinates from these files.

Best regards!

**Reply by DD-ZHAO** [reply]
Dear @david-ncu2019 
Thank you for your reply! I extracted geometryGeo.h5, which includes' height ',' incidenceAngle ',' slantRangeDistance ', and' waterMask ', but there is no latitude and longitude file.
Best regards!

**Reply by david-ncu2019** [reply]
Dear @DD-ZHAO ,

Have you solved the problem?

I'm also working with HYP3 products, and I found that `saveqgis.py` can export the time-series and velocity with coordinates.

**Reply by DD-ZHAO** [reply]
Dear @david-ncu2019,
Thank you for sharing. It seems that saveqgis.py can convert velocity.h5 into a QGIS recognizable shp file, but it seems that there are no attributes in the shp. I found that savegdal.py can convert .h5 to geotiff format, which may be more convenient to use.

**Reply by gabrielaquintana77** [reply]
Hi everyone, I used save_gdal.py but this dont give the correct lan lon!!!

**Reply by ditafaith** [reply]
> Hi everyone, I used save_gdal.py but this dont give the correct lan lon!!! 

Did you use hyp3 to obtain interferograms?

**Comment by gabrielaquintana77** [comment]
Yes, I used HyP3!

**Reply by AlekseyDmitriev** [reply]
See the answer for https://github.com/insarlab/MintPy/discussions/1296
